Makefile

```diff
@@ -10,12 +10,18 @@ ifneq ($(USE_SYSTEM_MPACK),no)
 USE_SYSTEM_LUA := 1
 endif
 
+PLATFORM := $(shell uname -s)
+
 # Lua-related configuration
 MPACK_LUA_VERSION ?= 5.1.5
 MPACK_LUA_VERSION_NOPATCH = $(basename $(MPACK_LUA_VERSION))
 LUA_URL ?= https://lua.org/ftp/lua-$(MPACK_LUA_VERSION).tar.gz
 LUAROCKS_URL ?= https://github.com/keplerproject/luarocks/archive/v2.2.0.tar.gz
+ifeq ($(PLATFORM), Darwin)
+LUA_TARGET ?= macosx
+else
 LUA_TARGET ?= linux
```

Comment:
looking again, it seems like LUA_TARGET was intended to be set by the caller or else default to "linux".

So adding the `uname` call seems questionable, because it would presumably fail on Windows. Why not pass LUA_TARGET from the caller (lua-client in your case specifically)?

Reply:
> An alternative to trying to have `make` work out of the box on osx is to update `README` with build instructions for osx users that amount to `LUA_TARGET=macosx make`. WDYT?

Yes. And if there's a LUA_TARGET for Windows that also helps :)


> not following your point about the caller being `lua-client` -- `LUA_TARGET` here as I understand it is just about building the local lua in `.deps` for a local build.

oops, right. I didn't inspect where it was used.

Reply:
@justinmk I've similarly updated https://github.com/neovim/lua-client/pull/55